### DISCUSSION TITLE
Find the editor for a Tab

### DISCUSSION THREAD

**Discussion by jogibear9988** [microsoft/vscode-discussions#558] [Extension Development QnA] [answered]
Hello,

I want to create a custom treeview for a custom editor. I now found I could use the Tabs API to find out wich "Tab" is active.
But I wanted to know if the "Tab" is a instance of my "CustomTextEditorProvider". Is there a way to do this?
Does a instance of a CustomTextEditorProvider, know it's own Tab?

Cause I could not use the ActiveTextEditor, cause CustomTextEditorProvider is no TextEditor

**Comment by JYC-99** [comment]
Methods like "revealCustomDocument" or "onCustomDocumentDidReveal" on the interface vscode.CustomEditorProvider could be useful. Unfortunately, I did not find any clues to resolve it after exploring the doc site and samples for days.

**Comment by ArturoDent** [comment, marked as answer]
The `Tab.input` says this:

> Defines the structure of the tab i.e. text, notebook, custom, etc. Resource and other useful properties are defined on the tab kind.

So what does `yourTab.input` hold for a customEditor?
